presto-spi/src/main/java/com/facebook/presto/spi/type/DoubleType.java

```diff
@@ -76,7 +76,8 @@ public boolean equalTo(Block leftBlock, int leftPosition, Block rightBlock, int
     @Override
     public long hash(Block block, int position)
     {
-        return AbstractLongType.hash(block.getLong(position, 0));
+        // convert to canonical NaN if necessary
+        return AbstractLongType.hash(doubleToLongBits(longBitsToDouble(block.getLong(position, 0))));
```

Comment:
I was thinking there might be a simple mask for this, but then I looked at the code for doubleToLongBits... wow